virt_who/testing.py

```diff
@@ -1124,11 +1124,18 @@ def vw_start_thread(
         return data, tty_output, rhsm_output
 
     def vw_start(self, cli=None, timeout=900, exp_send=1, exp_loopnum=0, oneshot=False, event=None, web_check=True, exp_error=False):
-        for i in range(3):
+        for i in range(4):
             data, tty_output, rhsm_output = self.vw_start_thread(cli, timeout, exp_send, exp_loopnum, oneshot, event, exp_error)
             if data['is_429'] == "yes":
                 wait_time = 60*(i+3)
-                logger.warning("429 code found, try again after %s seconds..." % wait_time)
+                logger.warning("429 code found, re-register virt-who host and try again after %s seconds..." % wait_time)
+                register_config = self.get_register_config()
+                register_type = register_config['type']
+                self.runcmd("\cp -f /etc/rhsm/rhsm.conf /root/rhsm429.conf", self.ssh_host())
+                self.system_unregister(self.ssh_host())
+                self.system_register_config(self.ssh_host(), register_type, register_config)
+                self.system_register(self.ssh_host(), register_type, register_config)
+                self.runcmd("mv /root/rhsm429.conf /etc/rhsm/rhsm.conf", self.ssh_host())
```

Comment:
why do we need to backup /etc/rhsm/rhsm.conf? 
`system_register_config()` will re-generate the rhsm.conf  according to stage or satellite, and `system_unregister` will be called by `rhsm_recovery` in this function.

Reply:
By testing, some cases, like tc_1045 will update the configurations (proxy_hostname, proxy_port) in /etc/rhsm/rhsm.conf, which makes re-register failed due to wrong proxy, so we need using system_register_config() to call rhsm_recovery to make sure the host can register successfully. 

And because of the rhsm_recovery,  next steps will be failed due to configuration missing. So we need to backup rhsm.conf at first and then recover it.

About why we need the system_unregister(), it's because if the code find the certification is right and subscription status is good, it will not unregister and register again, but here we just want the host can be re-register.

Reply:
1. `system_unregister` has already been included in `rhsm_recovery`
2.  `mv /root/rhsm429.conf /etc/rhsm/rhsm.conf` should be checked 'Yes' or 'No' as well

Reply:
Yes and removed ```system_unregister```and updated to ```mv -f /root/rhsm429.conf /etc/rhsm/rhsm.conf```.
Thanks!!!